### DISCUSSION TITLE
Can Fossify SMS Messenger import (XML) messages ?

### DISCUSSION THREAD

**Discussion by jehoshua7** [FossifyOrg/General-Discussion#513] [Q&A] [answered]
My SMS messages are backed up using "SMS Backup & Restore", see https://www.synctech.com.au/sms-backup-restore/
They are in XML format. Can Fossify SMS Messenger import an XML file please ? All contacts referred to in the XML backup file are already imported into the Contacts app, as a .vcf file , so there shouldn't be any unrelated contacts.

**Comment by Aga-C** [comment]
There's no possibility currently. If you'd like, you can create a [feature request](https://github.com/FossifyOrg/Messages/issues/new/choose) for it, but I can't guarantee at this moment that we will accept it for development.

**Reply by jehoshua7** [reply]
Okay, thanks.

**Comment by antodc** [comment, marked as answer]
@jehoshua7 You can use "SMS Backup & Restore" to restore the messages, and then use Fossify SMS Messenger to read them. Am I missing something?

**Reply by jehoshua7** [reply]
@antodc - Oh I had no idea that Fossify SMS messenger can read them after an import by  "SMS Backup & Restore", thanks. I thought they would use different databases. I was trying to steer away from 'google' apps on a new (de-googled) phone, so am very interested in Fossify, F-Droid, Aurora store, etc.

Yet the  "SMS Backup & Restore" app is downloadable from their site, so I don't have to go through a Google store process, so now I am assuming that the  "SMS Backup & Restore" app has none of the tracking,etc.

Your solution is a good one, thank you.

**Reply by Echolon** [reply]
I think it should be clarified, that "SMS Backup & Restore" is another app: https://www.synctech.com.au/sms-backup-restore/

**Comment by jehoshua7** [comment]
The format from SMS Backup and Restore is .XML. I see the Fossify/Messages format is .JSON. Will checkout some .XML to .JSON converters. Have exported messages from Fossify/Messages to inspect the format, as stuck on app permissions with restoring the .XML within SMS backup and Restore.

**Reply by antodc** [reply]
@jehoshua7 Maybe I was unclear. I do not think you need a XML/JSON converter.

Try the following steps:

1) Open SMS Backup and Restore
2) Restore your messages (the command is under the burger menu on the top left of the screen)
3) Open Fossify Messages

Do you see your restored messages in Fossify Messages?

**Reply by antodc** [reply]
EDIT Before point 1 above you should have created a backup of your messages with SMS Backup and Restore, if that was not already clear.

**Comment by jehoshua7** [comment]
@antodc - Your instructions were very clear. The problem was with SMS backup & Restore, a permissions problem. All permissions were granted except the "SMS" one, and I was unable to "Allow" that. The msg was "App was denied access". Therefor, I will follow up with the authors of that App.

Also, yes I did create a backup initially, it was from another phone. I did also mange to get some Python code going to convert from .XML to .JSON format. I do already have Fossify/messages going. These additional messages are from an old phone (Samsung S2) , so there is no rush. The contacts are loaded okay.

Thanks for your help.  :)

**Comment by Echolon** [comment]
Hi,

my Silence backup is > 16 MB and I it breaks and fails into a dark screen to me. This is the only thing holding me back from a switch to Fossify...

**Reply by naveensingh** [reply]
How did you import your messages? If you still have your old device, you can export/import using Fossify Messages or SMS Backup & Restore as mentioned in the [above comment](https://github.com/orgs/FossifyOrg/discussions/513#discussioncomment-9205277).

**Reply by Echolon** [reply]
@naveensingh thanks - I just did this with SMS Backup. Though, same story, when I import (the now just 2.80 MB file 🤔) it ends up in a black screen (though I see the android interaction buttons and top bar).

**Reply by naveensingh** [reply]
Do you see the black screen when importing your messages using SMS Backup & Restore, or do you see it in Fossify Messages _after_ your messages were imported? To be clear, the process is:

1.  Export messages using your chosen app from your old device
2. Import messages **using the same app** on your new device
3. Open Fossify Messages and wait until the progress bar at the top disappears

**Reply by Echolon** [reply]
I do see it in Fossify Messages _after_ / _while_ my messages were / are imported.

It is the same device. I am just switching apps.

Btw I never saw any progress bar in Fossify Messages

**Reply by Echolon** [reply]
I'm not sure if I understand your steps. Does it imply, that Fossify Messages will automatically read the messages once imported in the old app?

**Reply by naveensingh** [reply]
> I'm not sure if I understand your steps. Does it imply, that Fossify Messages will automatically read the messages once imported in the old app?

Yes, messages are stored in the system. 

> It is the same device. I am just switching apps.

In that case, you don't need to import/export at all. Just switch to Fossify Messages.

**Reply by Echolon** [reply]
> In that case, you don't need to import/export at all. Just switch to Fossify Messages.

That for sure did not work. Important contacts do not have any messages at all :/ Though, some contacts it did work actually.

I'm sorry, I don't want to be annoying here. I really admire the work on this as I rely on SMS. The migration just don't work for me so far :/

**Reply by naveensingh** [reply]
I am not familiar with Silence, but its encryption may be the problem here. Do you see those messages when you switch to other messaging apps like QUIK or Google Messages? If yes, please file a bug report with as much detail as possible: https://github.com/FossifyOrg/Messages/issues/new?template=bug_report.yml

**Reply by Echolon** [reply]
I guess that might be it... though I wonder why the import not work. I exported in plain text of course.

No, I don't see it in my default SMS app :/ Just very old ones.... Damn it :/
